src/core/mac/mac.cpp

```diff
@@ -728,6 +729,7 @@ void Mac::HandleAckTimer(void)
 
             if (mScanChannels == 0 || mScanChannel > kPhyMaxChannel)
             {
+                otPlatRadioSetPanId(NULL, mPanId);
```

Comment:
I can see how this would change the radio panid filter to 0xFFFF at start of scan, but when does it get set back to the regular PANID for the network?